modules/isIterable.js

```diff
@@ -0,0 +1,7 @@
+function isIterable(obj) {
+    // checks for null and undefined
+    if (obj == null) {
+      return false;
+    }
+    return typeof obj[Symbol.iterator] === 'function';
```

Comment:
We can't safely assume that `Symbol` or `Symbol.iterator` is defined, because that will (needlessly!) break compatibility with older environments that only know ES3. Instead, we would use feature detection and then create an internal `getIterator` function that simply returns `undefined` in environments where these objects don't exist. You can see an example of this practice in `modules/keys.js`, which uses `Object.keys` but only if it exists.

A similar, but more subtle, consideration applies to `typeof x === 'function'`. In some environments, this expression doesn't work as intended. This is why you should use `_.isFunction` instead whenever possible. `isFunction(x)` can also be minified much more compactly than `typeof x === 'function'`.

So this line should have read like this:

```suggestion
    return isFunction(getIterator(obj));
```